qa/tasks/cephfs/test_acls.py

```diff
@@ -0,0 +1,104 @@
+from textwrap import dedent
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+
+# TODO: perhaps move the code to prepare xfstests-dev repo to a separate file
+# so that we it can be reused by other/new testsuites
+class TestACLs(CephFSTestCase):
+    def setUp(self):
+        CephFSTestCase.setUp(self)
+        self.prepare_xfstests_dev()
+
+    def prepare_xfstests_dev(self):
+        # TODO: get rid of the following before merging
+        #self.repo_path = '/home/rishabh/repos/xfstests-dev'
+        self.get_xfstests_dev()
+        self.write_local_config()
+        self.install_xfstests_dev_deps()
+
+        self.mount_a.client_remote.run('sudo useradd fsgqa', omit_sudo=False)
+        self.mount_a.client_remoe.run('sudo groupadd fsgqa', omit_sudo=False,
+                                      check_status=False)
+        self.mount_a.client_remote.run('sudo useradd 123456-fsgqa',
+                                       omit_sudo=False)
+
+        self.mount_a.client_remote.run('make', cwd=self.repo_path)
+        self.mount_a.client_remote.run('sudo make install',
+            cwd=self.repo_path, omit_sudo=False)
+
+    def get_xfstests_dev(self):
+        """
+        Clone xfstests_dev repository. If already present, update it.
+        """
+        from os import getcwd
+        from os import stat
+        from os.path import join
+        from subprocess import call
+        from sys import version_info as sys_version_info
+
+        # TODO: make sure that repo is not cloned for every test. it should
+        # happen only once.
+        repourl = 'git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git'
+        self.repo_path = join(self.mount_a.client_remote.mkdtemp(), 'xfstests-dev')
+        self.mount_a.run_shell('%s %s %s' % ('git clone', repourl, self.repo_path))
+
+    def write_local_config(self):
+        from os.path import join
+        from teuthology.misc import sudo_write_file
+
+        self.get_test_and_scratch_dirs_ready()
+        mon_socket_addr = self.get_a_mon_socket_addr()
+        self.test_dev = mon_socket_addr + ':/' + self.test_dirname
+        self.scratch_dev = mon_socket_addr + ':/' + self.scratch_dirname
+
+        xfstests_config_contents = dedent('''\
+                export FSTYP=ceph
+                export TEST_DEV={}
+                export TEST_DIR={}
+                #export SCRATCH_DEV={}
+                #export SCRATCH_MNT={}
```

Comment:
This will test kcephfs. Are you also looking to run tests vs. ceph-fuse?

Reply:
Yep. But if this PR is really good I can start testing it on teuthology and get the test working for FUSE simultaneously on a separate PR so that this gets done faster.

What about `SCRATCH_DEV` and `SCRATCH_MNT`? They are commented right now because I can't get the tests running with them uncommented.